src/Cache.php

```diff
@@ -90,23 +42,24 @@ public function read($file)
 
     public function removeLegacyTags(array $data): array
     {
-        foreach (self::$lowestTags as $lowestPackage => $settings) {
-            $lowestVersion = $settings['version'];
-            $replacedPackages = $settings['replaces'];
-            if (!isset($data['packages'][$lowestPackage][$lowestVersion])) {
-                continue;
-            }
-            foreach ($data['packages'] as $package => $versions) {
-                if ($package !== $lowestPackage && !in_array($package, $replacedPackages, true)) {
+        if (!$this->symfonyRequire || !isset($data['packages']['symfony/symfony'])) {
+            return $data;
+        }
+        $symfonyVersions = $data['packages']['symfony/symfony'];
+
+        foreach ($data['packages'] as $name => $versions) {
+            foreach ($versions as $version => $package) {
+                if ('symfony/symfony' !== $name && !isset($symfonyVersions[preg_replace('/^(\d++\.\d++)\..*/', '$1.x-dev', $version)]['replace'][$name])) {
```

Comment:
Would not it be better to put two if and put the comparison `'symfony/symfony' === $name` in first with a `continue2;` ?

Reply:
the comparison is using `!==`, not `===`, so this wouldn't work.

Reply:
I meant, explode the if in two parts and change the `!==` by `===`

Reply:
I'm sorry, I don't understand, please suggest the patch you have in mind.

Reply:
Like : 
```
if ('symfony/symfony' === $name) {
   continue2;
}
if (!isset($symfonyVersions[preg_replace('/^(\d++\.\d++)\..*/', '$1.x-dev', $version)]['replace'][$name])) {
   // ....
}
```

Reply:
But this would *not* be doing the same.

Reply:
Indeed, i just understood 😅